cfme/tests/configure/test_proxy.py

```diff
@@ -2,69 +2,35 @@
 
 from cfme import test_requirements
 from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.ec2 import EC2Provider
 from cfme.cloud.provider.gce import GCEProvider
 from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.conf import cfme_data
-from cfme.utils.conf import credentials
-from cfme.utils.generators import random_vm_name
-from cfme.utils.log import logger
-from cfme.utils.net import find_pingable
 from cfme.utils.ssh import SSHClient
-from cfme.utils.virtual_machines import deploy_template
-from cfme.utils.wait import TimedOutError
 from cfme.utils.wait import wait_for
 
 pytestmark = [
     pytest.mark.tier(3),
-    pytest.mark.provider([AzureProvider, GCEProvider], scope='module'),
+    pytest.mark.provider([AzureProvider, GCEProvider, EC2Provider], scope='module'),
     pytest.mark.usefixtures('setup_provider_modscope'),
     test_requirements.appliance
 ]
 
 
 @pytest.fixture(scope='module')
-def proxy_machine():
-    """ Deploy vm for proxy test
-
-    This fixture uses for deploy vm on provider from yaml and then receive it's ip
-    After test run vm deletes from provider
-    """
-    try:
-        data = cfme_data.proxy_template
-        proxy_port = data.port
-        vm = deploy_template(
-            data.provider,
-            random_vm_name('proxy'),
-            template_name=data.template_name
-        )
-    except AttributeError:
-        msg = 'Missing data in cfme_data.yaml, cannot deploy proxy'
-        logger.exception(msg)
-        pytest.skip(msg)
-
-    try:
-        found_ip, _ = wait_for(
-            find_pingable,
-            func_args=[vm],
-            fail_condition=None,
-            delay=5,
-            num_sec=300
-        )
-    except TimedOutError:
-        msg = 'Timed out waiting for reachable proxy VM IP'
-        logger.exception(msg)
-        pytest.skip(msg)
-
-    yield found_ip, proxy_port
-    vm.delete()
+def proxy_machine(utility_vm):
+    utility_vm_ip, utility_vm_root_password = utility_vm
+    yield (utility_vm_ip, utility_vm_root_password,
+           cfme_data.utility_vm.proxy.port)
```

Comment:
_Optional_ This fixture is not really necessary, I think the utility_vm data should be folded into the utility_vm fixture (i.e. `utility_vm` also `yields` `cfme_data.utility_vm`). Having the utility_vm yield a small class with all the necessary info may be an option. I'm marking this comment optional because I know you've already refactored this stuff a bunch and we can do these things in follow up PRs.

Reply:
I refactored this a  bit.